services/scan/src/store.ts

```diff
@@ -322,100 +394,72 @@ export class Store {
+    const electionRow = this.client.one(
+      'select precinct_selection from election'
+    ) as { precinct_selection: string } | undefined;
+
+    const rawPrecinctSelection = electionRow?.precinct_selection;
+
+    if (!rawPrecinctSelection) {
+      // precinct selection is undefined when there is no election
+      return undefined;
+    }
+
+    const precinctSelectionParseResult = safeParseJson(
+      rawPrecinctSelection,
+      PrecinctSelectionSchema
+    );
+
+    if (precinctSelectionParseResult.isErr()) {
+      throw new Error('Unable to parse stored precinct selection.');
+    }
+
+    return precinctSelectionParseResult.ok();
   }
 
   /**
    * Sets the current precinct `scan` is accepting ballots for. Set to
    * `undefined` to accept from all precincts (this is the default).
    */
   setPrecinctSelection(precinctSelection?: PrecinctSelectionType): void {
-    this.setConfig(ConfigKey.PrecinctSelection, precinctSelection);
+    if (this.hasElection()) {
+      if (!precinctSelection) {
+        this.client.run('update election set precinct_selection = null');
+      } else {
+        this.client.run(
+          'update election set precinct_selection = ?',
+          JSON.stringify(precinctSelection)
+        );
+      }
+    } else {
+      throw new Error('Cannot set precinct selection without an election.');
+    }
   }
 
   /**
    * Gets the current polls state (open, paused, closed initial, or closed final)
    */
   getPollsState(): PollsStateType {
-    return this.getConfig(
-      ConfigKey.PollsState,
-      'polls_closed_initial',
-      PollsStateSchema
-    );
-  }
-
-  /**
-   * Sets the current polls state
-   */
-  setPollsState(pollsState: PollsStateType): void {
-    this.setConfig(ConfigKey.PollsState, pollsState);
-  }
-
-  /**
-   * Gets a config value by key.
-   */
-  private getConfig<T>(key: ConfigKey, schema: z.ZodSchema<T>): T | undefined;
-  private getConfig<T>(
-    key: ConfigKey,
-    defaultValue: T,
-    schema: z.ZodSchema<T>
-  ): T;
-  private getConfig<T>(
-    key: ConfigKey,
-    defaultValueOrSchema: z.ZodSchema<T> | T,
-    maybeSchema?: z.ZodSchema<T>
-  ): T | undefined {
-    debug('get config %s', key);
-
-    const row = this.client.one<[string]>(
-      'select value from configs where key = ?',
-      key
-    ) as { value: string } | undefined;
-
-    let defaultValue: T | undefined;
-    let schema: z.ZodSchema<T>;
-
-    if (maybeSchema) {
-      defaultValue = defaultValueOrSchema as T;
-      schema = maybeSchema;
-    } else {
-      schema = defaultValueOrSchema as z.ZodSchema<T>;
-    }
-
-    if (typeof row === 'undefined') {
-      debug('returning default value for config %s: %o', key, defaultValue);
-      return defaultValue;
-    }
-
-    const result = safeParseJson(row.value, schema);
+    const electionRow = this.client.one('select polls_state from election') as
+      | { polls_state: PollsStateType }
```

Comment:
```suggestion
      | { polls_state: string }
```

Reply:
I am assuming this is related to your previous comment - if we type it as `string`, then typescript expects some kind of validation.